src/gc.c

```diff
@@ -10,79 +10,148 @@
 // will wait for the next GC, to allow the space to be reused more
 // efficiently. default = on.
 #define FREE_PAGES_EAGER
-
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include <assert.h>
-#ifdef USE_MMAP
-# include <sys/mman.h>
-# include <malloc.h>
-#endif
 #include "julia.h"
 #include "julia_internal.h"
-
-#ifdef _P64
-# ifdef USE_MMAP
-#  define GC_PAGE_SZ 16384//bytes
-# else
-#  define GC_PAGE_SZ 12288//bytes
-# endif
-#else
-#define GC_PAGE_SZ  8192//bytes
+#ifndef _OS_WINDOWS_
+#include <sys/mman.h>
+#ifdef _OS_DARWIN_
+#define MAP_ANONYMOUS MAP_ANON
+#endif
 #endif
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-typedef struct _gcpage_t {
-    char data[GC_PAGE_SZ];
+#pragma pack(push, 1)
```

Comment:
why do you forceably un-align all of these data structures (including overriding the attempted `__attribute__((aligned (64))) ` below. although aligning to 64-bytes was probably a bit overkill too)?